File: code/community/1306402371709964289/deepgram-diarization-pricing.go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"net/http"
	"os"
)

func main() {
	apiKey := os.Getenv("DEEPGRAM_API_KEY")
	audioURL := os.Getenv("AUDIO_URL")

	// API endpoint
	url := "https://api.deepgram.com/v1/listen"

	// Prepare payload
	payload := map[string]interface{}{
		"url":     audioURL,
		"diarize": true,
	}
	jsonPayload, _ := json.Marshal(payload)

	// Create a new request
	req, err := http.NewRequest("POST", url, bytes.NewBuffer(jsonPayload))
	if err != nil {
		fmt.Println("Error creating request:", err)
		return
	}

	// Set headers
	req.Header.Set("Authorization", fmt.Sprintf("Token %s", apiKey))
	req.Header.Set("Content-Type", "application/json")

	// Perform the request
	client := &http.Client{}
	resp, err := client.Do(req)
	if err != nil {
		fmt.Println("Error making API call:", err)
		return
	}
	defer resp.Body.Close()

	// Output the response
	buf := new(bytes.Buffer)
	buf.ReadFrom(resp.Body)
	fmt.Println(buf.String())
}

File: code/community/1306402371709964289/deepgram-diarization-pricing.rs
use std::env;
use reqwest::{Client, Error};

#[tokio::main]
async fn main() -> Result<(), Error> {
    // Setting up environment variables
    let api_key = env::var("DEEPGRAM_API_KEY").expect("DEEPGRAM_API_KEY not set");
    let audio_url = env::var("AUDIO_URL").expect("AUDIO_URL not set");
    
    // Create a client
    let client = Client::new();

    // Make the request
    let response = client.post("https://api.deepgram.com/v1/listen")
        .header("Authorization", format!("Token {}", api_key))
        .header("Content-Type", "application/json")
        .query(&["diarize", "true"]) // Enable diarization
        .json(&serde_json::json!({ "url": audio_url }))
        .send()
        .await?;

    // Print the response text
    let response_text = response.text().await?;
    println!("{}", response_text);

    Ok(())
}